crates/rspack_plugin_javascript/src/visitors/swc_visitor/provide.rs

```diff
@@ -0,0 +1,168 @@
+
+    // println!("obj_expr_sym: {:?}", obj_expr.as_call().unwrap().callee);
+
+    for module_name in module_path.iter().skip(1) {
+      let member_expr = MemberExpr {
+        span,
+        obj: Box::new(obj_expr),
+        prop: MemberProp::Computed(ComputedPropName {
+          span,
+          expr: Box::new(Expr::Lit(Lit::Str(Str {
+            span,
+            value: module_name.to_string().into(),
+            raw: None,
+          }))),
+        }),
+      };
+
+      obj_expr = Expr::Member(member_expr);
+      // println!("obj_expr_sym: {:?}", obj_expr.as_member().unwrap().obj);
+    }
+
+    // println!("obj_expr_sym: {:?}", obj_expr.as_call().unwrap().callee);
+    obj_expr
+  }
+
+  fn create_call_expr(&self, span: Span, module_path: &str) -> CallExpr {
+    // println!("module_path: {}", module_path);
+    // println!("span: {:?}", span);
+
+    CallExpr {
+      span,
+      callee: Callee::Expr(Box::new(Expr::Ident(Ident::new("require".into(), span)))),
+      args: vec![ExprOrSpread {
+        spread: None,
+        expr: Box::new(Expr::Lit(Lit::Str(Str {
+          span,
+          value: module_path.to_string().into(),
+          raw: None,
+        }))),
+      }],
+      type_args: Default::default(),
+    }
+  }
+
+  fn get_nested_identifier_name(&self, member_expr: &MemberExpr) -> String {
+    let mut identifier_name = String::new();
+
+    fn build_identifier_name(member_expr: &MemberExpr, identifier_name: &mut String) {
+      match &*member_expr.obj {
+        Expr::Member(nested_member_expr) => {
+          build_identifier_name(nested_member_expr, identifier_name);
+        }
+        Expr::Ident(ident) => {
+          if !identifier_name.is_empty() {
+            identifier_name.push_str(".");
+          }
+          identifier_name.push_str(&ident.sym.to_string());
+        }
+        Expr::This(_) => {
+          if !identifier_name.is_empty() {
+            identifier_name.push_str(".");
+          }
+          identifier_name.push_str("this");
+        }
+        _ => {}
+      }
+
+      if let Some(ident_prop) = member_expr.prop.as_ident() {
+        identifier_name.push_str(".");
+        identifier_name.push_str(&ident_prop.sym.to_string());
+      }
+    }
+
+    build_identifier_name(member_expr, &mut identifier_name);
+    identifier_name
+  }
+}
+
+impl<'a> VisitMut for ProvideBuiltin<'a> {
+  fn visit_mut_expr(&mut self, expr: &mut Expr) {
+    *expr = match expr {
+      Expr::Ident(ident) => self.handle_ident(ident),
+      Expr::Member(member_expr) => self.handle_member_expr(member_expr),
+      _ => expr.clone(),
+    };
+
+    expr.visit_mut_children_with(self);
+  }
+
+  fn visit_mut_module_items(&mut self, node: &mut Vec<ModuleItem>) {
+    for (_key, module_paths) in self.opts {
+      // println!("key: {:?}", key);
+      // println!("module_paths: {:?}", module_paths);
+
+      let import_decl = ImportDecl {
+        span: DUMMY_SP,
+        specifiers: vec![],
```

Comment:
What's the usage of the `ImportDecl`? I saw it has no specifiers.

Reply:
Initially I thought we needed to mutate the Import Declarations too. However, after the current implementation, I don't think we need it anymore

Reply:
I see. I'm not sure I understand correctly.

For config like
```js
{
  $: ["jquery", "default"]
}
```

Instead of transforming like
```js
$('div')

// => 

import jquery from 'jquery'
jquery.default('div')
```

the current PR is doing something like

```js
$('div')

// => 

require('jquery').default('div')
```

Reply:
@dhruvkelawala Could you tell me that is my understanding correct? Wonderful work by the way.

Reply:
Yes, it's working as you described. I think webpack does it in the same way, doesn't it? Am I missing something?

Reply:
You are right. Maybe we should try to achieve the results similar to webpack with ES module imports. Do you have any pointers how we can achieve that?

Reply:
Okay 👍